department-of-veterans-affairs/caseflow
# Div/appeals 49715 a

Resolves [Task Actions: Other motion: Assign To Team](https://jira.devops.va.gov/browse/APPEALS-49715)

# Description
Created a modal to assign teams for correspondence. The modal contains a organization dropdown that gets dynamically filled with an organization array and instructions text field, and a cancel and submit (Assign task) button.

## Acceptance Criteria

## Testing Plan


1. Log into Caseflow as a supervisor user
2. Go to "Your Queue"
3.  Click "Correspondence Cases" in the "Switch Views" drop down
4.  Click any veteran in the "Veteran Details" column
5.  Select any Correspondence Type and click Save Changes
6.  Scroll to the bottom and click "Create Record"
7.  Click Continue to go to the second step
8.  Under the "Tasks not related to an Appeal" Click "+Add Task"
9. In the New Task modal that renders add and "Other Motion" task and add any test text in the instructions
10. Scroll to the bottom and click "Continue."
11. Again, scroll to the bottom and click "Submit," then "Confirm" in the pop-up modal.
12. Navigate to details page for that correspondence.
13. Under the **Correspondence and Appeals Tasks** tab,  at the bottom of the first page, there is a "Tasks not related to an appeal", section find the Other Motion task you created.
14. From the actions dropdown, select **Assign to team** 
15. Make sure the cancel and X are working as intended.
16. From the dropdown, assign a team, write some instructions, and submit the form. See that those assigned to and task instructions are updated correctly.
17. Also verify that assigned organization should not be there in the dropdown.
18. Verify the success and warning banners


# Frontend
## User Facing Changes

 BEFORE|AFTER
 ---|---

<img width="820" alt="Screenshot 2024-08-28 at 2 04 06 PM" src="https://github.com/user-attachments/assets/b874d71b-aadb-436a-84b8-8f456765ff9c">
<img width="1426" alt="Screenshot 2024-08-28 at 2 04 20 PM" src="https://github.com/user-attachments/assets/ee47c00f-2c4c-4e69-8ae9-91cd42186e4e">
<img width="1308" alt="Screenshot 2024-08-28 at 2 04 33 PM" src="https://github.com/user-attachments/assets/926ec8a6-01a2-4526-a2e3-6ebe7d76023e">
<img width="1408" alt="Screenshot 2024-08-30 at 9 28 43 AM" src="https://github.com/user-attachments/assets/13cde417-0349-4d55-818d-2456d71d88da">



## Storybook Story
*For Frontend (Presentation) Components*

# Backend
## Database Changes
*Only for Schema Changes*


## Integrations: Adding endpoints for external APIs

# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?

## Monitoring, Logging, Auditing, Error, and Exception Handling Checklist

### Monitoring

### Logging

### Auditing

### Error Handling

### Exception Handling
